devdoc/rc_ptr.md

```diff
@@ -0,0 +1,34 @@
+# `rc_ptr` requirements
+
+## Overview
+
+`rc_ptr` is a module that encapsulates reference-counted pointers.
+
+## Exposed API
+```c
+typedef void (*RC_PTR_FREE_FUNC)(void* context);
+
+typedef struct RC_PTR_TAG
+{
+    const void* ptr;
+    RC_PTR_FREE_FUNC free_func;
+} RC_PTR;
+
+THANDLE_TYPE_DECLARE(RC_PTR);
+
+#define PRI_RC_PTR "p"
+
+#define RC_PTR_VALUE(rc) ((rc)->ptr)
+
+#define RC_PTR_AS_VOIDPTR(rc) (((rc) == NULL) ? NULL : (rc)->ptr)
+
+MOCKABLE_FUNCTION(, THANDLE(RC_PTR), rc_ptr_create_with_move_memory, const void*, ptr, RC_PTR_FREE_FUNC, free_func);
```

Comment:
>with_move_memory [](http://example.com/codeflow?start=51&length=16)

We're not really moving memory. It's more like create with onwership transfer, which is truly create in this case. with_move_memory made me think that we're copying some memory, which would imply a size too.

Reply:
I've seen the term `with_move_memory` used when we want to create with ownership transfer. e.g,: `rc_string_create_with_move_memory`, `CONSTBUFFER_CreateWithMoveMemory`. 

So that's why I named it that.